internal/util/diagtree/diagtree.go

```diff
@@ -180,20 +180,23 @@ func (m *Node) severity() string {
 // The severity of a node.
 //
 // Nodes with their own (non-None) severity are always displayed on their own level.
-type Severity string
+type Severity struct{ s string }
 
-const (
-	None   Severity = ""
-	Info   Severity = "`🟢`"
-	Warn   Severity = "`🟡`"
-	Danger Severity = "`🔴`"
+var (
+	None   = Severity{""}
+	Info   = Severity{"`🟢`"}
+	Warn   = Severity{"`🟡`"}
+	Danger = Severity{"`🔴`"}
```

Comment:
This change prevents the previous issue from compiling. Before this change, I had mistakenly put:
```go
		msg.SetDescription("type changed from %q to %q", oldType, newType)
```

Here `"type changed from %q to %q"` was the severity literal (of type `Severity`), `oldType` was the description and `newType` its argument. This was wrong, but would compile.

With this change, that will be a compile error since a `string` is no longer a valid literal for `Severity`.